src/Core/Asset/GeometryData.hpp

```diff
@@ -214,21 +234,74 @@ class RA_CORE_API GeometryData : public AssetData
     inline bool hasPolyhedra() const;
 
     /// Return true if the object has vertex normals.
-    inline bool hasNormals() const;
+    [[deprecated( "Use hasAttribData( std::string name ) instead." )]] inline bool
+    hasNormals() const;
 
     /// Return true if the object has vertex tangent vectors.
-    inline bool hasTangents() const;
+    [[deprecated( "Use hasAttribData( std::string name ) instead." )]] inline bool
+    hasTangents() const;
 
     /// Return true if the object has vertex bitangent vectors.
-    inline bool hasBiTangents() const;
+    [[deprecated( "Use hasAttribData( std::string name ) instead." )]] inline bool
+    hasBiTangents() const;
 
     /// Return true if the object has vertex texture coordinates.
-    inline bool hasTextureCoordinates() const;
+    [[deprecated( "Use hasAttribData( std::string name ) instead." )]] inline bool
+    hasTextureCoordinates() const;
 
     /// Return true if the object has MaterialData.
     inline bool hasMaterial() const;
     /// \}
 
+    /**
+     *
+     * @tparam Container
+     * @param name
+     * @return Get container base on name (lock).
+     * @warning If AttribHandle corresponding to name doesn't exist, it's created
+     * and return. By using this method, user has read-write access to data, data is lock, when
+     * done call attribDataUnlock( std :: std::string name )
+     */
+    template <typename Container>
+    inline Container& getAttribDataWithLock( std::string name );
```

Comment:
Consider 
```suggestion
    inline Container& getAttribDataWithLock( const std::string& name );
```
instead (same for all the other methods).

Reply:
thx